cpp/src/endpoint.cpp

```diff
@@ -23,12 +23,22 @@
 
 namespace ucxx {
 
+static std::shared_ptr<Worker> getWorker(std::shared_ptr<Component> workerOrListener)
+{
+  auto worker = std::dynamic_pointer_cast<Worker>(workerOrListener);
+  if (worker == nullptr) {
+    auto listener = std::dynamic_pointer_cast<Listener>(workerOrListener);
```

Comment:
The type system will not save you if you pass a non worker-or-listener in here and you'll get some kind of bad de-reference. Should we do `if (listener == nullptr) { throw ...; }` ?

Reply:
I guess one can't use `std::variant` because the type of `parent` is just `Component`.

Reply:
Good catch, fixed in https://github.com/rapidsai/ucxx/pull/50/commits/0a876bf40d11fda391d1b5f8708dabb2e6174212. I think your comment about `std::variant` is also right, it would be useful if we were passing the parent when it was already cast into its final form, but that is unfortunately not the case here.